### ISSUE TITLE
[BUG] False positive "missing document end" error when directive (%) is present

### ISSUE THREAD

**Issue by andrei-korshikov** [adrienverge/yamllint#560]
Consider the following yaml:
```
%YAML 1.2
---
hello: world!
...
```
With `document-end: enable` you get `2:1  error  missing document end "..."  (document-end)`.

That's because the current token (`---`) is `DocumentStartToken` and previous token (`%YAML 1.2`) is `DirectiveToken`, but yamllint [belives](https://github.com/adrienverge/yamllint/blob/b92fc9cb3133673f284ccdef76446673bd81c7a8/yamllint/rules/document_end.py#L106) that accepted previous tokens are [`DocumentEndToken` or `StreamStartToken`](https://github.com/adrienverge/yamllint/blob/b92fc9cb3133673f284ccdef76446673bd81c7a8/yamllint/rules/document_end.py#L100) only.

**Comment by adrienverge** [comment]
Hello Andrei, thanks for the bug report! I can confirm the problem with yamllint 1.30.0.

Contributions are welcome :+1: (a fix in `yamllint/rules/document_end.py` + a non-regression test in `tests/rules/test_document_end.py`)

**Comment by tarannum-2002** [comment]
Hey @adrienverge, I'd like to work on this issue if I get some more information on it.

**Comment by DimitriPapadopoulos** [comment]
@tarannum-2002 All the information is already there:
```
$ cat << EOF > foobar.yaml
%YAML 1.2
---
hello: world!
...
EOF
$ 
$  cat <<EOF > foobar.yaml 
rules:
  document-end: enable
EOF
$ 
$ yamllint -c config.yaml foobar.yaml 
foobar.yaml
  2:1       error    missing document end "..."  (document-end)

$
